agent/config/config.go

```diff
@@ -399,6 +419,7 @@ type ServiceDefinition struct {
 	TaggedAddresses   map[string]ServiceAddress `mapstructure:"tagged_addresses"`
 	Meta              map[string]string         `mapstructure:"meta"`
 	Port              *int                      `mapstructure:"port"`
+	Ports             ServicePorts              `mapstructure:"ports"`
```

Comment:
Avoiding making copies
```suggestion
	Ports             *ServicePorts              `mapstructure:"ports"`
```

Reply:
@srahul3 `ServicePorts` is just a slice of `ServicePort` which is already a pointer and won't be copied.